karbon/karbon_cluster_registration_service_test.go

```diff
@@ -513,6 +526,68 @@ func TestKarbonGetK8sRegistrationList(t *testing.T) {
 	validateK8sClusterRegistrationList(t, response)
 }
 
+func TestKarbonGetK8sRegistrationListEmpty(t *testing.T) {
+	interceptor := khttpclient.NewInterceptor(http.DefaultTransport)
+	creds := testhelpers.CredentialsFromEnvironment(t)
+	nkeClient, err := NewKarbonAPIClient(creds, WithRoundTripper(interceptor))
+	require.NoError(t, err)
+
+	kctx := mock.NewContext(mock.Config{
+		Mode: keploy.MODE_TEST,
+		Name: t.Name(),
+	})
+	// returns type K8sCreateClusterRegistrationResponse
+	response, err := nkeClient.ClusterRegistrationOperations.GetK8sRegistrationList(kctx)
+	assert.NoError(t, err)
+	expected := K8sClusterRegistrationListResponse{
+		Data:     []*K8sClusterRegistration{},
+		Total:    0,
+		PageSize: 20,
+		Offset:   0,
+	}
+	validateEmptyK8sClusterRegistrationList(t, response, expected)
+}
+
+func TestKarbonGetK8sRegistrationListPaginated(t *testing.T) {
+	interceptor := khttpclient.NewInterceptor(http.DefaultTransport)
+	creds := testhelpers.CredentialsFromEnvironment(t)
+	nkeClient, err := NewKarbonAPIClient(creds, WithRoundTripper(interceptor))
+	require.NoError(t, err)
+
+	kctx := mock.NewContext(mock.Config{
+		Mode: keploy.MODE_TEST,
+		Name: t.Name(),
+	})
+	// returns type K8sCreateClusterRegistrationResponse
+	response, err := nkeClient.ClusterRegistrationOperations.GetK8sRegistrationList(kctx)
```

Comment:
should we create registration and then do a get for completeness instead of assuming existing setup